### DISCUSSION TITLE
[3차원 그래픽스] 시야각에 대한 질문

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#100] [Q&A] [answered]
게임 프로그래밍을 위한 3차원 그래픽스 2.4.3절(51p~)에 있는 투영 행렬의 유도 부분을 공부하던중 모르는 것이 생겼습니다.

> 먼저 y'를 계산하자. 그림 2.20-(a)는 뷰 프러스텀의 단면을 보여주는데, v와v'는 각각 (y,z)와 (y',z')로 표기되었다. 가상의 투영 평면을 -cot( fovy / 2) 에 배치했는데, 이로인해 y'의 범위는 [-1,1]로 제한된다. 그림 2.20-(a)에 보인 바와 같이 y'는 닮은꼴 삼각형(similar triangles)을 이용해 구할 수 있다.  

> y' = -cot fovy / 2 * y / z ( * 이 책에서는 dot으로 표시되어있습니다.)

(비례식의 경우 따로 작성하지 않겠습니다.)

닯은꼴로 비례식을 쓰는데 왜 절대값을 쓰지 않고 -cot fovy / 2 라는 크기를 그대로 가져가는 걸까요.

왠지 기초적인 수학이 부족해서 드는 질문일수도 있다는 느낌이 드네요.

**Comment by Woon-2** [comment, marked as answer]
반대로, 절댓값을 써야하는 이유가 뭘까요?   
비례식에 음수 스칼라의 이용을 인정하기만 하면, 아무 문제 없을텐데요.   

-4 : -2 = -2 : -1은 우리가 흔히 비례식에서 이용하는 성질인 내항의 곱과 외항의 곱이 같다는 성질을 만족합니다.   
닮음비도 1/2로 우리가 원하는 값이 잘만 나옵니다.

지평을 넓혔을 뿐입니다.   

2차 방정식의 판별식 D = b<sup>2</sup>-4ac에 대해 D < 0인 경우
근이 존재하지 않는다라고 말하던 걸,   
실제로 존재하지 않는 수(a.k.a. 허수)까지 도입해서 허근이 존재한다고 말하는 것처럼요.

닮음은 길이의 비를 이용하는 것이니 무조건 절댓값을 씌워야 한다!   
하고 그대로 가봅시다.   
어차피 우리의 목적은 길이를 구하는 것이 아니라 좌표를 구하는 것이기 때문에   
절댓값을 씌워서 구한 길이에 다시 부호를 붙여야 하죠.   
비효율적입니다.

이런 부분에 대해선 수학자들도 딱히 이견을 낼 것 같진 않지만,   
우린 수학자가 아니니까, 수학적 진리를 규명하기 보다 수학을 도구로 이용할 줄 알아야 합니다.
